File: 111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        if not root.left and not root.right:
            return 1
        l = self.minDepth(root.left)
        r = self.minDepth(root.right)
        if l > 0 and r > 0:
            return min(l, r) + 1
        elif l > 0 and r == 0:
            return l + 1
        else:
            return r + 1

File: 111. Minimum Depth of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (root === null)
        return 0;
    if (root.left === null && root.right === null)
        return 1;
    let l = minDepth(root.left), r = minDepth(root.right);
    if (l > 0 && r > 0)
        return Math.min(l, r) + 1;
    else
        return l > 0 ? l + 1 : r + 1;
};
